pkg/vpcmodel/subnetsConnectivity.go

```diff
@@ -0,0 +1,289 @@
+package vpcmodel
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/common"
+)
+
+// VPCsubnetConnectivity captures allowed connectivity for subnets, considering nacl and pgw resources
+type VPCsubnetConnectivity struct {
+	// computed for each node (subnet), by iterating its ConnectivityResult for all relevant VPC resources that capture it
+	AllowedConns map[string]*ConfigBasedConnectivityResults
+	// combined connectivity - considering both ingress and egress per connection
+	AllowedConnsCombined map[string]map[string]*common.ConnectionSet
+	cloudConfig          *CloudConfig
+}
+
+type IPbasedConnectivityResult struct {
+	IngressAllowedConns map[*common.IPBlock]*common.ConnectionSet
+	EgressAllowedConns  map[*common.IPBlock]*common.ConnectionSet
+}
+
+type ConfigBasedConnectivityResults struct {
+	IngressAllowedConns map[string]*common.ConnectionSet
+	EgressAllowedConns  map[string]*common.ConnectionSet
+}
+
+const (
+	subnetKind                = "Subnet"
+	errUnexpectedTypePeerNode = "unexpected type for peerNode in computeAllowedConnsCombined"
+)
+
+func subnetConnLine(subnet string, conn *common.ConnectionSet) string {
+	return fmt.Sprintf("%s : %s\n", subnet, conn.String())
+}
+
+func (c *ConfigBasedConnectivityResults) String() string {
+	res := "Ingress: \n"
+	for n, conn := range c.IngressAllowedConns {
+		res += subnetConnLine(n, conn)
+	}
+	res += "Egress: \n"
+	for n, conn := range c.EgressAllowedConns {
+		res += subnetConnLine(n, conn)
+	}
+
+	return res
+}
+
+// print AllowedConns (not combined)
+func (v VPCsubnetConnectivity) printAllowedConns() {
```

Comment:
Having both pointer and non-pointer receivers for the same type is usually a bad idea

Reply:
right.. fixed